pkg/ddl/notifier/subscribe.go

```diff
@@ -255,7 +272,31 @@ func (n *DDLNotifier) processEventForHandler(
 	return nil
 }
 
+// Stop stops the background loop.
+func (n *DDLNotifier) Stop() {
+	// If the notifier is not started, the cancel function is nil.
+	if n.cancel != nil {
```

Comment:
If caller will call `Close` in a case that hasn't `Start`, it's possible that `Start` races with `Close`? And the execution is complicated.

The only special case which calls `Close` is `Domain.Close`. I think we can delete it. Let `Domain`'s `ctx` control the owner to exit, and then `OnRetireOwner` will be called. It's more clear.

Reply:
We can't close the session context in OnRetireOwner, because the node can be the owner again. But we only init the ddl notifier once with the sctx, so if the sctx is closed, there is no way to restart the ddl notifier again.